swingset/src/main/java/com/nqadmin/swingset/formatting/SSFormattedTextField.java

```diff
@@ -86,59 +86,76 @@
+	 * 
+	 * This class should perform validation AND call setValue() which will trigger a RowSet update.
 	 * <p>
-	 * See https://docs.oracle.com/javase/8/docs/api/javax/swing/JFormattedTextField.html
-	 * <p>
-	 * More on input verifiers here: https://www.drdobbs.com/jvm/java-better-interfaces-via-jformattedtex/224700979?pgno=1
-	 * Some discussion on dates and validation here: https://docs.oracle.com/javase/tutorial/uiswing/misc/focus.html#inputVerification
+	 * See https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/javax/swing/JFormattedTextField.html
 	 */
-	// TODO: This still needs some work.
 	public class FormattedTextFieldVerifier extends InputVerifier {
 
 		@Override
 		public boolean verify(final JComponent input) {
 
 			boolean result = true;
 			Object value = null;
+			
+			// Set indicator to suppress duplicate property change event if
+			// a call is made to setValue()
+			verifyingText = true;
 
 			if (input instanceof SSFormattedTextField) {
 				logger.debug("{}: Instance of SSFormattedTextField.", () -> getColumnForLog());
+				
 				final SSFormattedTextField ssftf = (SSFormattedTextField) input;
-				AbstractFormatter formatter = ssftf.getFormatter();
 				String formattedText = ssftf.getText();
+				
+				AbstractFormatter formatter = ssftf.getFormatter();
 				logger.debug("Formatter is: " + formatter + ".");
+				if (formatter==null) {
+					logger.warn("Null formatter encountered for formatted text field.");
+				}
+				
 				if (formatter!=null && formattedText!=null && !formattedText.isEmpty()) {
-					
 					try {
 						value = formatter.stringToValue(formattedText);
-						// Apparently formatter.stringToValue(formattedText) accomplished the same thing as commitEdit(),
+						// Apparently formatter.stringToValue(formattedText) accomplishes the same thing as commitEdit(),
 						// but this approach lets us know if the formatter is null.
-						//ssftf.commitEdit(); 
 					} catch (ParseException pe) {
 						logger.warn(getColumnForLog() + ": String of '" + formattedText + "' generated a Parse Exception at " + pe.getErrorOffset() + ".", pe);
 						result = false;
-						// We're not going to call setValue(null) if result is false.
+						// We're not going to call setValue(null) if result is false, rather we'll keep the
+						// focus in the current field.
 					}
 				} else {
-					logger.debug("Null formatter or text. Setting value to null.");
-// TODO: Deal with null formatter and non-null text.					
-					// Value is already null.
+				// value variable is set to null be default, but make a log entry
+					logger.debug("Null formatter, empty string, or null text.");
 				}
 
-				// now perform custom checks
+				// now perform custom validation/range checks
 				if (result) {
 					result = validateField(value);
 				}
 
-				// update text color for negatives
+				// update text color for negatives, where applicable
 				if (result) {
 					updateTextColor(value);
```

Comment:
I think you mentioned this elsewhere, but I agree that globally making negatives red is a very specific thing to do and seems like something that a formatter should be able to handle. I was thinking of a static method that could take a JFormattedTextField and the value, but I'm not sure that helps much. How about `decorateComponent()`? We could also deprecate it, although I do like making negatives red in our app :)

Reply:
I was looking for a method name that give the usage idea. I'm good with it being there. In the future can set things up so that this kind of app UI could be plugged in, and the default would be what's there now.